packages/bsky/src/services/moderation/status.ts

```diff
@@ -0,0 +1,272 @@
+// This may require better organization but for now, just dumping functions here containing DB queries for moderation status
+
+import { AtUri } from '@atproto/syntax'
+import { PrimaryDatabase } from '../../db'
+import {
+  ModerationEvent,
+  ModerationSubjectStatus,
+} from '../../db/tables/moderation'
+import {
+  ACKNOWLEDGE,
+  REVIEWOPEN,
+  MUTE,
+  REVIEWCLOSED,
+  REPORT,
+  REVIEWESCALATED,
+  REVERT,
+  TAKEDOWN,
+  ESCALATE,
+} from '../../lexicon/types/com/atproto/admin/defs'
+import { ModerationEventRow, ModerationSubjectStatusRow } from './types'
+
+const actionTypesImpactingStatus = [
+  ACKNOWLEDGE,
+  REPORT,
+  ESCALATE,
+  REVERT,
+  TAKEDOWN,
+  MUTE,
+]
+
+// TODO: How do we handle revert? for "revert" event we will have a reference event id that is being reversed
+// We will probably need a helper that can take a list of events and compute the final state of the subject
+// That helper will have to be invoked here with all events up until the point where the reverted event was created
```

Comment:
Is the idea that history would be replayed with the reverted event missing?  I wonder if we should think on this a bit and see if we can make our lives a little simpler for reverts if we can.

Reply:
yeah I'm not too clear on this either. The current implementation carries the concept of `status impacting event`which helps us find the last event that could have changed the status of the subject BEFORE the reverted event and "replays" that event to produce the side-effect. but it gets really hairy if we think about potential multiple reverts and removing effects of events since the reverted event. 
Maybe we can limit this to not allow reverting specific event and ONLY allow reverting "the last status impacting event" which seems practical but I need to think through the potential edge cases around this.